File: bmi.java
public class bmi {	
	public static void main(String[] args){
		if(args.length!=2)
		{
			System.out.println("Usage: <command> <weight [kg]> <height [m]>");
			return;
		}
		else
		{
			try
			{
				double weight = Double.parseDouble(args[0]);
				double height = Double.parseDouble(args[1]);
				double bmi = weight / (height * height);
				System.out.println("Your BMI is:\t"+bmi);
				String output;
				if(bmi <= 18.5)
				{
				    output="You're underweight.";
				}
				else if(bmi <= 25)
				{
				    output="You're healthy.";
				}
				else if(bmi <= 30)
				{
				    output="You're overweight.";
				}
				else
				{
				    output="Holy McShizzle, you're freaking whale!";
				}
				System.out.println(output);
			}
			catch(NumberFormatException e)
			{
				System.out.println("Commandline arguments must be pure numerical!");
			}
		}
	}
}

File: bmi.scala
object bmi {
  def main(args: Array[String]): Unit = {
    if (args.length != 2) {
      println("Usage: <command> <weight [kg]> <height [m]>")
      return
    }
    try {
      val Array(weight, height) = args.map(_.toDouble)
      val bmi = weight / math.pow(height, 2.0)

      printf("Your BMI is: %.2f\n", bmi)

      bmi match {
        case x if x <= 18.5 => println("You're underweight.")
        case x if x <= 25.0 => println("You're healthy.")
        case x if x <= 30.0 => println("You're overweight.")
        case _ => println("Holy McShizzle, you're freaking whale!")
      }
    } catch {
      case _: NumberFormatException => println("Arguments must be numbers.")
    }
  }
}
